src/Microsoft.DotNet.Cli.Utils/Command.cs

```diff
@@ -57,11 +57,18 @@ public static Command Create(string executable, string args)
 
         private static void ResolveExecutablePath(ref string executable, ref string args)
         {
-            var fullExecutable = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, executable + Constants.ExeSuffix));
-
-            if (File.Exists(fullExecutable))
+            foreach (string suffix in Constants.RunnableSuffixes)
```

Comment:
We discussed this and the summary is ShouldUseCmd does a no-op for this case, hence the bug.